src/transactions/transactionBytes.js

```diff
@@ -203,132 +148,110 @@ function getEmptyBytesForTransaction(transaction) {
 }
 
 /**
- * @method createTransactionBuffer
- * @param transaction Object
- * @return {buffer}
- */
+* A utility class to get transaction byteSizes
+*
+* @class Transaction
+* @param {Object} transaction
+* @constructor
+*/
 
-function createTransactionBuffer(transaction) {
-	function assignHexToTransactionBytes(partTransactionBuffer, hexValue) {
-		const hexBuffer = Buffer.from(hexValue, 'hex');
-		for (let i = 0; i < hexBuffer.length; i++) {
-			partTransactionBuffer.writeByte(hexBuffer[i]);
-		}
-		return partTransactionBuffer;
-	}
+export class Transaction {
+	constructor(transaction) {
+		this.byteSizes = byteSizes;
+		this.transaction = transaction;
 
-	/**
-	 * @method createEmptyTransactionBuffer
-	 * @param assetSize number
-	 * @return {buffer}
-	 */
+		this.transactionType = Buffer.alloc(this.byteSizes.TYPE);
```

Comment:
Prefer allocating when creating the Buffer: `Buffer.alloc(this.byteSizes.TYPE, transaction.type)`